avm/res/network/virtual-hub/main.bicep

```
@@ -0,0 +1,190 @@
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Required. Address-prefix for this VirtualHub.')
+param addressPrefix string
+
+@description('Optional. Flag to control transit for VirtualRouter hub.')
+param allowBranchToBranchTraffic bool = true
+
+@description('Optional. Resource ID of the Express Route Gateway to link to.')
+param expressRouteGatewayId string = ''
+
+@description('Optional. Resource ID of the Point-to-Site VPN Gateway to link to.')
+param p2SVpnGatewayId string = ''
+
+@description('Optional. The preferred routing gateway types.')
+@allowed([
+  'ExpressRoute'
+  'None'
+  'VpnGateway'
+  ''
+])
+param preferredRoutingGateway string = ''
+
+@description('Optional. VirtualHub route tables.')
+param routeTableRoutes array = []
+
+@description('Optional. ID of the Security Partner Provider to link to.')
+param securityPartnerProviderId string = ''
+
+@description('Optional. The Security Provider name.')
+param securityProviderName string = ''
+
+@allowed([
+  'Basic'
+  'Standard'
+])
+@description('Optional. The sku of this VirtualHub.')
+param sku string = 'Standard'
+
+@description('Optional. List of all virtual hub route table v2s associated with this VirtualHub.')
+param virtualHubRouteTableV2s array = []
+
+@description('Optional. VirtualRouter ASN.')
+param virtualRouterAsn int = -1
+
+@description('Optional. VirtualRouter IPs.')
+param virtualRouterIps array = []
+
+@description('Required. Resource ID of the virtual WAN to link to.')
+param virtualWanId string
+
+@description('Optional. Resource ID of the VPN Gateway to link to.')
+param vpnGatewayId string = ''
+
+@description('Optional. Route tables to create for the virtual hub.')
+param hubRouteTables array = []
+
+@description('Optional. Virtual network connections to create for the virtual hub.')
+param hubVirtualNetworkConnections array = []
+
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+var enableReferencedModulesTelemetry = false
+
+resource avmTelemetry 'Microsoft.Resources/deployments@2023-07-01' = if (enableTelemetry) {
+  name: take('46d3xbcp.res.network-virtualhub.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}', 64)
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource virtualHub 'Microsoft.Network/virtualHubs@2022-11-01' = {
+  name: name
+  location: location
+  tags: tags
+  properties: {
+    addressPrefix: addressPrefix
+    allowBranchToBranchTraffic: allowBranchToBranchTraffic
```

Noticed a few missing params for basic functionality - think we can get this in? 

example:

```suggestion
    allowBranchToBranchTraffic: allowBranchToBranchTraffic
    azureFirewall: !empty(azureFirewallId): {
      id: azureFirewallId
    }
```

---

If added, please note the parameter is not existing & @ChrisSidebotham it should be `azureFirewallResourceId` not `azureFirewallId`

---

@ChrisSidebotham , I guess this can be resolved now